tests/various/struct_access.ys

```diff
@@ -0,0 +1,43 @@
+read_verilog -sv -dump_ast1 -dump_ast2 << EOF
+module dut();
+typedef struct packed {
+  logic a;
+  logic b;
+} struct_t;
+
+typedef struct packed {
+  struct_t c;
+} sub_struct_t;
+
+typedef struct packed {
+  sub_struct_t d;
+  sub_struct_t e;
+} sub_sub_struct_t;
+
+parameter sub_sub_struct_t P = 4'b1100;
+
+localparam sub_struct_t f = P.d;
+localparam sub_struct_t g = P.e;
+localparam struct_t h = f.c;
+localparam logic i = P.d.c.a;
+localparam logic j = P.d.c.b;
+localparam x = P.e;
+localparam y = x.c;
+localparam z = y.a;
+
+always_comb begin
+  assert(P == 4'b1100);
+  assert(f == 2'b11);
+  assert(g == 2'b00);
+  assert(h == 2'b11);
+  assert(i == 1'b1);
+  assert(j == 1'b1);
+  assert(x == 2'b00);
+  assert(y == 2'b00);
```

Comment:
It should be possible to access the fields of `x` and `y` given their implicit type. Can this be supported?

Reply:
Yes. I've added support for this and added it to tests.